app/assets/javascripts/categories_legend.js

```diff
@@ -1,4 +1,8 @@
 var CategoryLegend = {
+  color_map: {},
+  selected_values: [],
+  select2_selector: '#q_tags_filter',
```

Comment:
shouldn't be `CategoryLegend.prototype.select2_selector` ?

This whole class has to by instantiated in `map.js` and use then use this instance.